model.create does not have a lean option, and using toObject and toJSON does not produce similar typing

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

8.8.3

### Node.js version

22.11.0

### MongoDB server version

7.0

### Typescript version (if applicable)

4.9.5

### Description

While upgrading to the new 8.8.3 due to a dependabot alert, types broke for my app,
when doing a find, if not found create scenario and reusing the `let` variable, which has a type inferred already, as a LeanDocument (only exists on find/Query type) which is compatible with lots of other types previously, but is not anymore due to the addition of the __v to it, and not in `toObject` and `toJSON` which are my usual workarounds to get a type from a newly created Document (Document type)
Doing the following workarounds that i usually use does not work anymore, (except casting which i want to avoid):
- use `document.toObject` this used to produce a type identical
```ts
 let currentRecord = await recordModel.findById(recordId).lean().exec();

    if (!currentRecord) {
      let newRecord = {
      ... datapoints
      };

      const createdRecord = await recordModel.create(newRecord);

      currentRecord = createdRecord.toObject(); // used to work with no type errors in 8.7.3
    }
```

TS error using 8.8.3:
```
src/file.ts(253,7): error TS2322: Type 'T & Required<{ _id: ObjectId; }>' is not assignable to type '{ _id?: ObjectId; ... 38 more ... } & Required<...> & { ...; }'.
Property '__v' is missing in type 'T & Required<{ _id: ObjectId; }>' but required in type '{ __v: number; }'.
```
Output of toObject:
` T & Required<{
    _id: mongoose.Types.ObjectId;
}>`

Output of lean.exec which gives a raw-er datatype (which used to work fine):
`{
    _id?: mongoose.Types.ObjectId;
    ... 38 more ...;
} & Required<{
    _id: mongoose.Types.ObjectId;
}> & {
    __v: number;
}
`

### Steps to Reproduce

https://github.com/Amoodaa/mongoose-8-8-type-error

On `8.8.3`:
```bash
index.ts:43:5 - error TS2322: Type 'Foo & { _id: ObjectId; }' is not assignable to type '{ state: "on" | "off"; } & { _id: ObjectId; } & { __v: number; }'.
  Property '__v' is missing in type 'Foo & { _id: ObjectId; }' but required in type '{ __v: number; }'.

43     foundFoo = createdFoo.toObject(); // this errors
       ~~~~~~~~

  node_modules/mongoose/types/index.d.ts:143:13
    143     : T & { __v: number };
                    ~~~
    '__v' is declared here.


Found 1 error in index.ts:43
```

On `8.7.3`: no errors, check readme for more

### Expected Behavior

I would expect there would some kind of type similarity between the different lean versions? like toObject
A seamless way to get away from mongoose types and go back to my defined types, or at least type that is very compatible with that,
I've migrated from 6 -> 7 -> 8 recently and have faced many problems with mongoose typing solution, but its obviously better than being blind so i really appreciate your work on typings, thank you so much

--- a/test/types/document.test.ts
+++ b/test/types/document.test.ts
@@ -456,6 +456,6 @@ async function gh15077() {
 
     const createdFoo = await fooModel.create(newFoo);
 
-    foundFoo = createdFoo.toObject(); // this errors on 8.8.3
+    foundFoo = createdFoo.toObject();
   }
 }

--- a/test/types/document.test.ts
+++ b/test/types/document.test.ts
@@ -423,3 +423,39 @@ async function gh14876() {
   expectType<UserObjectInterface>(populatedCar.owner);
   expectType<Types.ObjectId>(depopulatedCar.owner);
 }
+
+async function gh15077() {
+  type Foo = {
+    state: 'on' | 'off';
+  };
+
+  const fooSchema = new Schema<Foo>(
+    {
+      state: {
+        type: String,
+        enum: ['on', 'off']
+      }
+    },
+    { timestamps: true }
+  );
+
+  const fooModel = model('foo', fooSchema);
+
+  let foundFoo = await fooModel
+    .findOne({
+      state: 'on'
+    })
+    .lean()
+    .exec();
+
+  if (!foundFoo) {
+    const newFoo = {
+      state: 'on'
+      // extra props but irrelevant
+    };
+
+    const createdFoo = await fooModel.create(newFoo);
+
+    foundFoo = createdFoo.toObject(); // this errors on 8.8.3
+  }
+}

--- a/types/document.d.ts
+++ b/types/document.d.ts
@@ -256,21 +256,21 @@ declare module 'mongoose' {
     set(value: string | Record<string, any>): this;
 
     /** The return value of this method is used in calls to JSON.stringify(doc). */
-    toJSON(options?: ToObjectOptions & { flattenMaps?: true, flattenObjectIds?: false }): FlattenMaps<Require_id<DocType>>;
-    toJSON(options: ToObjectOptions & { flattenObjectIds: false }): FlattenMaps<Require_id<DocType>>;
-    toJSON(options: ToObjectOptions & { flattenObjectIds: true }): ObjectIdToString<FlattenMaps<Require_id<DocType>>>;
-    toJSON(options: ToObjectOptions & { flattenMaps: false }): Require_id<DocType>;
-    toJSON(options: ToObjectOptions & { flattenMaps: false; flattenObjectIds: true }): ObjectIdToString<Require_id<DocType>>;
+    toJSON(options?: ToObjectOptions & { flattenMaps?: true, flattenObjectIds?: false }): FlattenMaps<Default__v<Require_id<DocType>>>;
+    toJSON(options: ToObjectOptions & { flattenObjectIds: false }): FlattenMaps<Default__v<Require_id<DocType>>>;
+    toJSON(options: ToObjectOptions & { flattenObjectIds: true }): ObjectIdToString<FlattenMaps<Default__v<Require_id<DocType>>>>;
+    toJSON(options: ToObjectOptions & { flattenMaps: false }): Default__v<Require_id<DocType>>;
+    toJSON(options: ToObjectOptions & { flattenMaps: false; flattenObjectIds: true }): ObjectIdToString<Default__v<Require_id<DocType>>>;
 
-    toJSON<T = Require_id<DocType>>(options?: ToObjectOptions & { flattenMaps?: true, flattenObjectIds?: false }): FlattenMaps<T>;
-    toJSON<T = Require_id<DocType>>(options: ToObjectOptions & { flattenObjectIds: false }): FlattenMaps<T>;
-    toJSON<T = Require_id<DocType>>(options: ToObjectOptions & { flattenObjectIds: true }): ObjectIdToString<FlattenMaps<T>>;
-    toJSON<T = Require_id<DocType>>(options: ToObjectOptions & { flattenMaps: false }): T;
-    toJSON<T = Require_id<DocType>>(options: ToObjectOptions & { flattenMaps: false; flattenObjectIds: true }): ObjectIdToString<T>;
+    toJSON<T = Default__v<Require_id<DocType>>>(options?: ToObjectOptions & { flattenMaps?: true, flattenObjectIds?: false }): FlattenMaps<T>;
+    toJSON<T = Default__v<Require_id<DocType>>>(options: ToObjectOptions & { flattenObjectIds: false }): FlattenMaps<T>;
+    toJSON<T = Default__v<Require_id<DocType>>>(options: ToObjectOptions & { flattenObjectIds: true }): ObjectIdToString<FlattenMaps<T>>;
+    toJSON<T = Default__v<Require_id<DocType>>>(options: ToObjectOptions & { flattenMaps: false }): T;
+    toJSON<T = Default__v<Require_id<DocType>>>(options: ToObjectOptions & { flattenMaps: false; flattenObjectIds: true }): ObjectIdToString<T>;
 
     /** Converts this document into a plain-old JavaScript object ([POJO](https://masteringjs.io/tutorials/fundamentals/pojo)). */
-    toObject(options?: ToObjectOptions): Require_id<DocType>;
-    toObject<T>(options?: ToObjectOptions): Require_id<T>;
+    toObject(options?: ToObjectOptions): Default__v<Require_id<DocType>>;
+    toObject<T>(options?: ToObjectOptions): Default__v<Require_id<T>>;
 
     /** Clears the modified state on the specified path. */
     unmarkModified<T extends keyof DocType>(path: T): void;

